tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -1099,12 +1113,14 @@ protected:
                files[slot] = merger.GetFile();
                trees[slot] = new TTree(treename.c_str(), treename.c_str());
                trees[slot]->ResetBit(kMustCleanup);
+               trees[slot]->SetAutoFlush(autoflush);
                // hack to call TTree::Branch on all variadic template arguments
                std::initializer_list<int> expander = {(trees[slot]->Branch(bnames[S].c_str(), &args), 0)..., 0};
                (void)expander; // avoid unused variable warnings for older compilers such as gcc 4.9
             }
             trees[slot]->Fill();
-            if (files[slot]->GetBytesWritten() >= cachesize) files[slot]->Write();
+            if (autoflush > 0 && trees[slot]->GetEntries() % autoflush == 0)
```

Comment:
If by 'buffer are written' you mean the 'Write' is triggered very often ... then I am very confused (because it should only triggered when the cluster is full to 30Mb of compressed data) ... something must still be wrong with the test itself ...